File: nodejs/poll-status.js
const API_KEY = process.env.MIDJOURNEY_API_KEY;
const BASE_URL =
  process.env.MIDJOURNEY_API_BASE_URL || "https://api.midjourney-api.com";

async function main() {
  if (!API_KEY) {
    throw new Error("Missing MIDJOURNEY_API_KEY");
  }

  const taskId = process.argv[2];
  if (!taskId) {
    throw new Error("Usage: npm run poll -- <task_id>");
  }

  const response = await fetch(`${BASE_URL}/midjourney/v1/job-status`, {
    method: "POST",
    headers: {
      "API-KEY": API_KEY,
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      taskIds: [taskId],
    }),
  });

  if (!response.ok) {
    throw new Error(`Request failed: ${response.status}`);
  }

  const data = await response.json();
  console.log(JSON.stringify(data, null, 2));
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: python/poll_status.py
import os
import sys

import requests


API_KEY = os.getenv("MIDJOURNEY_API_KEY")
BASE_URL = os.getenv("MIDJOURNEY_API_BASE_URL", "https://api.midjourney-api.com")


def main() -> int:
    if not API_KEY:
        print("Missing MIDJOURNEY_API_KEY")
        return 1

    if len(sys.argv) < 2:
        print("Usage: python3 python/poll_status.py <task_id>")
        return 1

    task_id = sys.argv[1]

    response = requests.post(
        f"{BASE_URL}/midjourney/v1/job-status",
        headers={
            "API-KEY": API_KEY,
            "Content-Type": "application/json",
        },
        json={"taskIds": [task_id]},
        timeout=30,
    )
    response.raise_for_status()
    print(response.json())
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
